src/core/time_arrays.jl

```diff
@@ -272,64 +459,220 @@ split_indices(idxs, ti) = idxs[1:ti - 1], idxs[ti], idxs[ti + 1:end]
 
 function Base.getindex(arr::TimestepArray{FixedTimestep{FIRST, STEP}, T, N, ti}, idxs::Union{FixedTimestep{FIRST, STEP, LAST}, AnyIndex}...) where {T, N, ti, FIRST, STEP, LAST}
 	idxs1, ts, idxs2 = split_indices(idxs, ti)
-	return arr.data[idxs1..., ts.t, idxs2...]
+	if isa(ts, Colon)
+		return arr.data[idxs1..., :, idxs2...]
+	else
+		return arr.data[idxs1..., ts.t, idxs2...]
+	end
 end
 
 function Base.getindex(arr::TimestepArray{VariableTimestep{TIMES}, T, N, ti}, idxs::Union{VariableTimestep{TIMES}, AnyIndex}...) where {T, N, ti, TIMES}
 	idxs1, ts, idxs2 = split_indices(idxs, ti)
-	return arr.data[idxs1..., ts.t, idxs2...]
+	if isa(ts, Colon)
+		return arr.data[idxs1..., :, idxs2...]
+	else
+		return arr.data[idxs1..., ts.t, idxs2...]
+	end
 end
 
 function Base.getindex(arr::TimestepArray{FixedTimestep{D_FIRST, STEP}, T, N, ti}, idxs::Union{FixedTimestep{T_FIRST, STEP, LAST}, AnyIndex}...) where {T, N, ti, D_FIRST, T_FIRST, STEP, LAST}
 	idxs1, ts, idxs2 = split_indices(idxs, ti)
-	t = Int(ts.t + (FIRST - TIMES[1]) / STEP)
-	return arr.data[idxs1..., t, idxs2...]
+	if isa(ts, Colon)
+		return arr.data[idxs1..., :, idxs2...]
+	else
+		t = Int(ts.t + (FIRST - TIMES[1]) / STEP)
+		return arr.data[idxs1..., t, idxs2...]
+	end
 end
```

Comment:
So I think there might be a different approach to to all of these functions where you've added the `isa(ts, Colon)...`

I _think_ instead we can change the method signature to be `idx::Union{FixedTimestep{FIRST, STEP, LAST}, AnyIndex_NonColon}` So that all of these  methods only get used when there is a timestep, not a colon.

Because all of these functions that have the call to `split_indices` are only supposed to be for when the index in the time position is a timestep, since we need to find which one it is and do `ts.t` or whatever the specific calculation is to get the t index. If the index in the time position is a colon, however, then I think that should just go to a generic method that just does `arr.data[idxs...]`. Thoughts?

Reply:
Great idea this worked!